src/server/api/routers/matchRequest.ts

```diff
@@ -455,83 +377,25 @@ export const matchRequestRouter = createTRPCRouter({
     });
   }),
 
-  // Users confirm that they want to match with the other user
+  // confirm a match between 2 different users
   confirmMatch: protectedProcedure
-    .input(z.object({ acceptId: z.string(), requestId: z.string() }))
+    .input(z.object({ userId1: userId_z, userId2: userId_z }))
     .mutation(async ({ ctx, input }) => {
-      const { acceptId, requestId } = input;
-      const r2 = await ctx.prismaPostgres.matchRequest.findUnique({
-        where: {
-          id: requestId,
-        },
+      const { userId1, userId2 } = input;
+      const matchedRequests = await ctx.prismaPostgres.matchRequest.findMany({
+        where: { userId: { in: [userId1, userId2] } },
       });
-      if (!r2)
+      if (matchedRequests.length !== 2) {
         throw new TRPCError({
           code: "BAD_REQUEST",
-          message: "Failed to match requests",
+          message: "Match request for both users not found",
         });
```

Comment:
I'm not sure if we want both users to have match requests in the lobby before one is allowed to accept the other's request

Reply:
oh i think ur right there is likely a race condition if we don't use queues here (if thats what you are saying)